realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -361,6 +371,57 @@ private static RealmCache getCache(String realmPath, boolean createIfNotExist) {
         return (E) refAndCount.localRealm.get();
     }
 
+    /**
+     * Synchronize all initial subscriptions to disk (if needed).
+     *
+     * If this fails for a new Realm file, the file will be deleted so a new attempt can be done later.
+     * Old Realm files will be left alone.
+     *
+     * This method is not threadsafe. Synchronization should happen outside it.
+     *
+     * @param realm Realm instance to synchronize instances for. It is safe to close this Realm if an exception is thrown.
+     * @param realmFileExistedOnDisk {@code true} if the file existed on disk before trying to open the Realm.
+     */
+    private static void synchronizeInitialSubscriptionsIfNeeded(Realm realm, boolean realmFileExistedOnDisk) {
+        if (!realmFileExistedOnDisk) {
```

Comment:
`realmFileExistedOnDisk` is set before creating anything at the entry of the function, but I agree the name isn't very descriptive at this point in time. I changed the name to `realmFileIsBeingCreated` and reversed the boolean logic in the places required.

The check is here because `initialData` is only ever called for typed Realms, not DynamicRealm.